core/src/main/kotlin/io/customer/sdk/communication/EventBus.kt

```diff
@@ -0,0 +1,56 @@
+package io.customer.sdk.communication
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.SharedFlow
+import kotlinx.coroutines.flow.filterIsInstance
+import kotlinx.coroutines.launch
+
+/**
+ * Event class to be used for communication between different modules.
+ */
+interface EventBus {
+    val flow: SharedFlow<Event>
+    fun publish(event: Event)
+    fun cancelAll()
+}
+
+/**
+ * Implementation of [EventBus] using [SharedFlow] for event handling.
+ * param scope: [CoroutineScope] to be used for event handling.
+ * param flow: [SharedFlow] to be used for event handling and buffer for replay.
+ */
+class EventBusImpl(
+    val scope: CoroutineScope = CoroutineScope(Dispatchers.Default + SupervisorJob()),
```

Comment:
We are already taking it in constructor so we can provide the test scoped one in tests, `DispatchersProvider` would have made sense if it was utilized by multiple components sharing the same dispatchers.

Reply:
But would still be an issue when we try to access `EventBus` from `SDKComponent` in tests relying on it because we won't be constructing EventBus in such cases I think